### PR TITLE
Core/Scripts/DB: Migrate specially scripted taxi NPC's to SmartAI

### PR DESCRIPTION
**Pull request by mysterious-static**
Updates NPCs Susurrus (17435), Protectorate Nether Drake (20903), Veronia (20162), Cassa Crimsonwing (23704), and Brazen (18725). 

Some notes: 
- For the quest associated with the Protectorate Nether Drake (On Nethery Wings), the quest still has some weird behavior. It's a little outside the scope of the PR, so I'll probably be opening a separate issue for that.
- Brazen had different behavior on retail from what was scripted, so that was also changed.

### CODE REVIEW

File: src/server/game/Spells/SpellMgr.cpp [hunk 1]
```diff
@@ -3762,6 +3762,9 @@ void SpellMgr::LoadSpellInfoCorrections()
                 break;
             // ENDOF ISLE OF CONQUEST SPELLS
             //
+            case 34905:
```
Comment (ccrs) [inline src/server/game/Spells/SpellMgr.cpp]: ermm hack xd

````SpellRange: (Id 1) "Self Only"````
Use castback or something around that

File: sql/updates/world/2016_01_07_npc_taxi.sql [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+(14,7959,9779,0,0,2,0,25853,1,0,0,0,'','Brazen - Show gossip menu text if player has item 25853'),
+(14,7959,9780,0,0,2,0,25853,1,0,1,0,'','Brazen - Show gossip menu text if player does not have item 25853'),
+(15,7959,0,0,0,2,0,25853,1,0,0,0,'','Brazen - Show gossip option 0 if player has item 25853');
+
+
+-- Create SmartAI for Brazen
+DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 18725);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(18725,0,0,1,62,0,100,0,7959,0,0,0,52,534,0,0,0,0,0,7,0,0,0,0,0,0,0,'Brazen - On Gossip Option 0 Selected - Activate Taxi Path 534'),
+(18725,0,1,0,61,0,100,0,0,0,0,0,72,0,0,0,0,0,0,7,0,0,0,0,0,0,0,'Brazen - On Linked Actions - Close Gossip');
```
Comment (ccrs) [inline sql/updates/world/2016_01_07_npc_taxi.sql]: new line

File: src/server/game/Spells/SpellMgr.cpp (hunk 1 above)
Comment (velinath) [inline src/server/game/Spells/SpellMgr.cpp]: ha, I am dumb, did not even see that in list of actions. Thanks. Will change and test.

**Comment by jackpoz** [comment]
did you test all the entries with all the conditions ingame ?

File: sql/updates/world/2016_01_07_npc_taxi.sql [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+(14,8082,9989,0,0,28,0,10652,0,0,1,0,'','Veronia - Show gossip menu text if player does not have quest 10652'),
+(14,8082,9990,0,0,28,0,10652,0,0,0,0,'','Veronia - Show gossip menu text if player has quest 10652'),
+(15,8082,0,0,0,28,0,10652,0,0,0,0,'','Veronia - Show gossip option 0 if player has quest 10652');
+-- Create SmartAI for Veronia
+DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 20162);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(20162,0,0,1,62,0,100,0,8082,0,0,0,85,34905,0,0,0,0,0,7,0,0,0,0,0,0,0,'Veronia - On Gossip Option 0 Selected - Invoker Cast \'Stealth Flight\''),
+(20162,0,1,0,61,0,100,0,0,0,0,0,72,0,0,0,0,0,0,7,0,0,0,0,0,0,0,'Veronia - On Linked Actions - Close Gossip');
+
```
Comment (MitchesD) [inline sql/updates/world/2016_01_07_npc_taxi.sql]: remove unneeded empty lines

**Comment by velinath** [comment]
Yes. Still waiting to check on changes that @ccrs mentioned though

**Comment by velinath** [comment]
Upon testing, that change worked. Squashed. I'll keep an eye out in case further changes are needed.

**Pull request merged**